src/server/req_delete.c

```diff
@@ -515,10 +557,22 @@ req_deletejob(struct batch_request *preq)
 
 			++preq->rq_refct;
 			preply->brp_un.brp_deletejoblist.tot_arr_jobs++;
+			
+			if (preq->rq_ind.rq_deletejoblist.rq_resume)
+				start = preq->rq_ind.rq_deletejoblist.subjobid_to_resume;
 
 			/* keep the array from being removed while we are looking at it */
 			parent->ji_ajinfo->tkm_flags |= TKMFLG_NO_DELETE;
-			for (i = parent->ji_ajinfo->tkm_start; i <= parent->ji_ajinfo->tkm_end; i += parent->ji_ajinfo->tkm_step) {
+			for (i = start; i <= parent->ji_ajinfo->tkm_end; i += parent->ji_ajinfo->tkm_step) {
+				end_time = time(NULL);
+				if ((end_time - begin_time) > 5) {
```

Comment:
please do not hardcode 5 - use a macro

Reply:
Thanks, I have added macro for this duration.